src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreJob.cs

```diff
@@ -108,6 +108,7 @@ internal sealed class SolutionRestoreJob : ISolutionRestoreJob
 
             // update instance attributes with the shared context values
             _nuGetProjectContext = jobContext.NuGetProjectContext;
+            _nuGetProjectContext.OperationId = request.OperationId;
```

Comment:
you dont need to get this operation id from request. You can generate one there itself

Reply:
Consider this case, some other nuget operation trigger a restore job, we want to understand this relation. then the other operation can set this OperationId to help us to know the parent.